R/calendar_range.R

```diff
@@ -0,0 +1,114 @@
+#' Create Semantic UI Calendar Range
+#'
+#' This is a wrapper for creating calendar ranges using Semantic UI calendars.
+#' It creates a form composed of two calendars. The selected range values are
+#' available under \code{input[[input_id]]}.
+#'
+#' @details The Semantic UI calendar range automatically pops up the end range
+#' calendar when changing the start range. Therefore events are only sent on
+#' changes done in the end range calendar.
+#'
+#' @param input_id Input name. Reactive value is available under \code{input[[input_id]]}.
+#' @param type Select from \code{'year'}, \code{'month'}, \code{'date'} and \code{'time'}.
+#' @param start_value Initial value of the calendar defining the start of the range.
+#' @param end_value Initial value of the calendar defining the end of the range.
+#' @param start_placeholder Text visible in the start calendar input when nothing is inputted.
+#' @param end_placeholder Text visible in the end calendar input when nothing is inputted.
+#' @param min Minimum allowed value in both calendars.
+#' @param max Maximum allowed value in both calendars.
+#'
+#' @rdname calendar_range
+#' @export
+#'
+calendar_range <- function(input_id, type = "date", start_value = NULL, end_value = NULL,
+                           start_placeholder = NULL, end_placeholder = NULL, min = NA, max = NA) {
+  if (!is.null(start_value)) start_value <- format(as.Date(start_value), "%Y/%m/%d")
+  if (!is.null(end_value)) end_value <- format(as.Date(end_value), "%Y/%m/%d")
+  if (!is.na(min)) min <- format(as.Date(min), "%Y/%m/%d")
+  if (!is.na(max)) max <- format(as.Date(max), "%Y/%m/%d")
+
+  create_cal_widget <- function(type, value, placeholder, min, max) {
```

Comment:
For readability, it's worth extracting this function outside of calendar_range()